integration_test/Bindings/Python/dialects/rtl.py

```diff
@@ -14,6 +14,16 @@
 
   i32 = IntegerType.get_signless(32)
 
+  # CHECK: !rtl.array<5xi32>
+  # CHECK: i32
+  # CHECK: 5
+  array_i32 = rtl.ArrayType.get(i32, 5)
+  print(array_i32)
+  # print(array_i32.element_type)
```

Comment:
@stellaraccident it seems the readonly properties are only working for me if I construct the object through the initializer, like below. Did I miss something with how I set this up? I also think I'm seeing the same behavior with ESI.

Reply:
Did you ever get this resolved?

Reply:
Not yet. I can just start with the constructor for now and we can add those later.

Reply:
Confirmed that I'm still seeing this behavior:

```
>>> import circt.esi
>>> c32 = circt.esi.ChannelType.get(i32)
>>> i32
Type(i32)
>>> c32
Type(!esi.channel<i32>)
>>> c32.inner
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: '_mlir.ir.Type' object has no attribute 'inner'
```

I think I'll just merge the new types without the readonly accessor fields for now.

Reply:
Ah, the `get` methods are returning an instance of the base class (by virtue of the MlirType caster). We should add a `def_factory` for defining such getters and it can internally call the derived class constructor. In python, this would typically call for the use of a classmethod, but that is awkward/overkill in c++.

Reply:
I pushed an update upstream which fixes this. Feel free to copy-pasta it locally until we get that landed: https://reviews.llvm.org/D102144